lib/tbot/botfs/fs_linux.go

```diff
@@ -62,45 +62,58 @@ const modeACLNone fs.FileMode = 0
 // missingSyscallWarning is used to reduce log spam when a syscall is missing.
 var missingSyscallWarning sync.Once
 
-// openSecure opens the given path for writing (with O_CREAT, mode 0600)
-// with the RESOLVE_NO_SYMLINKS flag set.
-func openSecure(path string, mode OpenMode) (*os.File, error) {
+// openSecure opens the given path for either reading or writing with the
+// RESOLVE_NO_SYMLINKS flag set.
+func openSecure(path string, flags OpenFlags) (*os.File, error) {
+	var mode uint64
+	if flags == ReadFlags {
+		// openat2() with nonzero mode will raise EINVAL unless O_CREATE or
+		// O_TMPFILE is set, so zero it out.
+		mode = 0
+	} else {
+		mode = uint64(DefaultMode.Perm())
+	}
```

Comment:
`mode` will be zero by default, so we can simplify this:

```suggestion
	var mode uint64
	if flags != ReadFlags {
		mode = uint64(DefaultMode.Perm())
	}
```